onnxruntime/test/ir/flatbuffer_utils_test.cc

```diff
@@ -0,0 +1,409 @@
+    fbs_tensors.push_back(fbs_tensor);
+  }
+
+  // TODO: might be 844 depending on whether it's 4 byte or 8 byte alignment
+  ASSERT_EQ(output_stream.tellp(), 840) << "Data written to the external file is incorrect.";
+  output_stream.close();
+  ASSERT_TRUE(output_stream.good()) << "Failed to close data file.";
+
+  auto fbs_tensors_offset = builder.CreateVector(fbs_tensors);
+  fbs::test::TestDataBuilder tdb(builder);
+  tdb.add_initializers(fbs_tensors_offset);
+  builder.Finish(tdb.Finish());
+  auto fb_data = builder.GetBufferSpan();
+
+  auto test_data = fbs::test::GetTestData(fb_data.data());
+  auto fbs_tensors2 = test_data->initializers();
+
+  // read
+  std::ifstream input_stream;
+  ExternalDataReader reader;
+  CreateReader("ExternalWriteReadBasicTest.bin", input_stream, reader);
+
+  std::vector<ONNX_NAMESPACE::TensorProto> loaded_initializers;
+  OrtFormatLoadOptions options;
+
+  for (const auto* fbs_tensor : *fbs_tensors2) {
+    ONNX_NAMESPACE::TensorProto initializer;
+    ASSERT_STATUS_OK(LoadInitializerOrtFormat(*fbs_tensor, initializer, options, reader));
+    loaded_initializers.emplace_back(std::move(initializer));
+    // also check that the loaded flatbuffer tensors have accurately written to the external_data_offset field
+    if (fbs_tensor->data_type() != fbs::TensorDataType::STRING && fbs_tensor->name()->str() != "tensor_32_small")
+    {
+      ASSERT_TRUE(fbs_tensor->external_data_offset() >= 0) << "external_data_offset is not set when we expect it to be set for tensor " << fbs_tensor->name()->str();
+    }
+    else
+    {
+      ASSERT_TRUE(fbs_tensor->external_data_offset() == -1) << "external_data_offset is set for string data when we expect it to not be set for tensor " << fbs_tensor->name()->str();
+      ASSERT_TRUE(fbs_tensor->raw_data() || fbs_tensor->string_data()) << "tensor has no data attached to it" << fbs_tensor->name()->str();
+    }
+  }
+
+  bool data_validated = true;
+
+  // initializers = expected, in the form of tensorproto
+  // loaded_initializers = actual, in the form of tensorproto
+  ASSERT_EQ(initializers.size(), loaded_initializers.size());
+
+  for (int i = 0; i < initializers.size(); i++) {
+    const auto& expected_initializer = initializers[i];
+    const auto& loaded_initializer = loaded_initializers[i];
+    // validate the loaded initializer
+    ASSERT_EQ(expected_initializer.name(), loaded_initializer.name());
+    ASSERT_EQ(expected_initializer.data_type(), loaded_initializer.data_type());
+    ASSERT_EQ_FB_TENSORPROTO_VECTORFIELD(expected_initializer, loaded_initializer, dims());
+    if (loaded_initializer.data_type() != ONNX_NAMESPACE::TensorProto_DataType_STRING) {
+      // extract expected tensor raw data
+      std::vector<uint8_t> expected_data;
+      Path model_path;
+      ASSERT_STATUS_OK(onnxruntime::utils::UnpackInitializerData(expected_initializer, model_path, expected_data));
```

Comment:
if they don't have external data, could simply be
```suggestion
      ASSERT_STATUS_OK(onnxruntime::utils::UnpackInitializerData(expected_initializer, expected_data));
```